smetrics/src/main/scala/com/evolutiongaming/smetrics/CollectorRegistry.scala

```diff
@@ -254,4 +259,132 @@ object CollectorRegistry {
       magnet: LabelsMagnetInitialized[A, B]
     ): Resource[F, B[Histogram[F]]] = delegate.histogramInitialized(prefixedName(name), help, buckets, labels)
   }
+
+  final class CachedRegistryException(message: String, cause: Throwable = null) extends RuntimeException(message, cause)
+
+  private def cached[F[_] : Concurrent](registry: CollectorRegistry[F]): Resource[F, CollectorRegistry[F]] = {
+
+    case class Entry(collector: Any, names: List[String], ofType: String)
+
+    for {
+      cache <- SerialRef.of[Resource[F, *], Map[String, Entry]](Map.empty)
```

Comment:
interesting choice :) 

could you please ensure that you remove elements upon release and not keep them forever?